packages/dev-server/src/MetroDevServer.ts

```diff
@@ -162,12 +168,24 @@ export async function bundleAsync(
     return await Promise.all(
       bundles.map(async (bundle: BundleOptions) => {
         const bundleOutput = await buildAsync(bundle);
-        const shouldBuildHermesBundle = await shouldBuildHermesBundleAsync(
+        const isHermesManaged = isEnableHermesManaged(expoConfig, bundle.platform);
+
+        const maybeInconsistentEngine = await maybeInconsistentEngineAsync(
           projectRoot,
-          bundle.platform
+          bundle.platform,
+          isHermesManaged
         );
+        if (maybeInconsistentEngine) {
+          throw new Error(
+            `Inconsistent Hermes settings between app config and ${bundle.platform} native project.\n` +
+              `Enabling Hermes in app.json or app.config.js: ${isHermesManaged.toString()}\n` +
+              `Please check the following native project settings if there are any inconsistencies:\n` +
+              '  - android/gradle.properties\n' +
+              '  - android/app/build.gradle\n'
+          );
```

Comment:
```suggestion
            `JavaScript engine configuration is inconsistent between ${configFileName) and ${bundle.platform} native project.\n` +
              `In ${configFileName}: Hermes is ${isHermesManaged ? 'enabled' : 'not enabled'}\n` +
              `In ${bundle.platform} native project: Hermes is ${isHermesManaged ? 'not enabled' : 'enabled'}\n`
              `Please check the following files for inconsistencies:\n` +
              '  - android/gradle.properties\n' +
              '  - android/app/build.gradle\n'
          );
```

what do you think about this? i think it reads a bit more clearly. `configFileName` refers to the other comment abou `getConfigFilePaths`